1-foundations/1-types-and-syntax/foundations-s1.ipynb

```diff
@@ -1,11 +1,1580 @@
+    "tuple_mix[1] = \"Marvin\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use tuples to get information and store them in variables\n",
+    "name = tuple_mix[1]\n",
+    "print('Variable name:', name, type(name))\n",
+    "\n",
+    "# Items copied from a tuple no longer need to be immutable\n",
+    "name = name.upper()\n",
+    "print('Variable name upper:', name, type(name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# If we type cast a tuple to a list it behaves as a list\n",
+    "list_from_tuple = list(tuple_mix)\n",
+    "list_from_tuple[1] = \"Marvin\"\n",
+    "print('Variable list_from_tuple:', list_from_tuple, type(list_from_tuple))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# While items in a tuple are immutable, the tuple can be overwritten\n",
+    "tuple_mix = tuple(list_from_tuple)\n",
+    "print('Variable tuple_mix:', tuple_mix, type(tuple_mix))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Container types - Sets\n",
+    "\n",
+    "| Class name | Full name  | Access                      | Occurrence  | Remarks |\n",
+    "|:---        |:---        | :---                        | :---        | :---\n",
+    "| set        | Set        | Test if item already in set | Rare        | A container that cannot hold duplicates |\n",
+    "\n",
+    "Rare but included it for completion. It's a container that cannot have duplicates. \n",
+    "\n",
+    "---\n",
+    "\n",
+    "**Demonstrate a set:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a set that stores all skills in a team:\n",
+    "team_skills = set()\n",
+    "print('Variable team_skills:', team_skills, type(team_skills))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define skillsets for person A, B and C\n",
+    "personA_skill1 = \"python\"\n",
+    "personB_skill1 = \"field-work\"\n",
+    "personC_skill1 = \"python\"\n",
+    "personC_skill2 = \"Excel\"\n",
+    "\n",
+    "# Add skillsets for person A, B and C\n",
+    "team_skills.add(personA_skill1)\n",
+    "team_skills.add(personB_skill1)\n",
+    "team_skills.add(personC_skill1)\n",
+    "team_skills.add(personC_skill2)\n",
+    "print('Variable team_skills:', team_skills, type(team_skills))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add a list of person D's skills to the set\n",
+    "personD_skills = [\"python\",\"field-work\",\"management\",\"accounting\"]\n",
+    "team_skills.update(personD_skills)\n",
```

Comment:
I tried using a string as an input in `.update()`,the result is that every unique letter is added to the set. It must happen because strings are containers too for Python. I don't think we should mention that at this point (but you can do it if you prefer), but remember to clarify that `update()` is not meant to be used with string values and `add()` should be the method for that

Reply:
good point, but I prefer to not add anything more on sets as they are so rare. If they in the future get advanced enough where they use sets they will be able to figure that difference. And probably not remember what I said in the very end of this session